coalib/misc/Caching.py

```diff
@@ -0,0 +1,193 @@
+import calendar
+import os
+import pickle
+import time
+
+from coalib.misc.Constants import caching_db, changed_files_db
+from coalib.output.Tagging import get_tags_dir as get_pickle_dir
+
+
+def get_cache_data_path(log_printer, filename):
+    """
+    Get the full path of ``filename`` present in the user's cache directory.
+
+    :param log_printer: A LogPrinter object to use for logging.
+    :param filename:    The file whose path needs to be expanded.
+    :return:            Full path of the file, assuming it's present in the
+                        user's config directory.
+    """
+    return os.path.join(get_pickle_dir(log_printer), filename)
+
+
+def delete_cache_files(log_printer, files=[caching_db, changed_files_db]):
+    """
+    Delete the cache files after displaying a warning saying the cache
+    is corrupted and will be removed.
+
+    :param log_printer: A LogPrinter object to use for logging.
+    :param files:       The list of files to be deleted.
+    :return:            True if all the given files were successfully deleted.
+                        False otherwise.
+    """
+    log_printer.warn("The caching database is corrupted and will "
```

Comment:
This warning is unrelated to "delete_cache_files", it should be in the `except` in `pickle_load` I think